source/agora/test/Base.d

```diff
@@ -1069,9 +1069,13 @@ public class TestAPIManager
             else
             {
                 // Send transaction to the first client
-                spendables.takeExactly(1)
-                    .map!(txb => txb.sign())
-                    .each!(tx => first_client.putTransaction(tx));
+                auto tx = spendables.takeExactly(1).map!(txb => txb.sign()).front;
+                first_client.putTransaction(tx);
+                // Wait for tx gossipping before setting time for block
+                client_idxs.each!(idx =>
+                    retryFor(this.clients[idx].hasTransactionHash(tx.hashFull()),
```

Comment:
As a side note, I'm starting to think we should have a `retryFor` that takes a total time (this can take 12 seconds without timing out if we have 6 clients).

Reply:
It could do them in parallel I suppose if we let retryFor take a client range.